client/extensions/woocommerce/state/sites/settings/email/actions.js

```diff
@@ -0,0 +1,361 @@
+	settings
+} );
+
+const mailchimpCampaignDefaultsSubmitFailure = ( siteId, { error } ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_CAMPAIGN_DEFAULTS_SUBMIT_FAILURE,
+	siteId,
+	error
+} );
+
+const mailchimpListsRequest = ( siteId ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_LISTS_REQUEST,
+	siteId
+} );
+
+const mailchimpListsRequestSuccess = ( siteId, lists ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_LISTS_REQUEST_SUCCESS,
+	siteId,
+	lists
+} );
+
+const mailchimpListsRequestFailure = ( siteId, { error } ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_LISTS_REQUEST_FAILURE,
+	siteId,
+	error
+} );
+
+const mailchimpSyncStatusRequest = ( siteId ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_SYNC_STATUS_REQUEST,
+	siteId
+} );
+
+const mailchimpSyncStatusRequestSuccess = ( siteId, syncStatus ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_SYNC_STATUS_REQUEST_SUCCESS,
+	siteId,
+	syncStatus
+} );
+
+const mailchimpSyncStatusRequestFailure = ( siteId, { error } ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_SYNC_STATUS_REQUEST_FAILURE,
+	siteId,
+	error
+} );
+
+const mailchimpResyncRequest = ( siteId ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_RESYNC_REQUEST,
+	siteId
+} );
+
+const mailchimpResyncRequestSuccess = ( siteId, syncStatus ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_RESYNC_REQUEST_SUCCESS,
+	siteId,
+	syncStatus
+} );
+
+const mailchimpResyncRequestFailure = ( siteId, { error } ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_RESYNC_REQUEST_FAILURE,
+	siteId,
+	error
+} );
+
+const mailchimpNewsletterSettingsSubmit = ( siteId ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_NEWSLETTER_SETTINGS_SUBMIT,
+	siteId
+} );
+
+const mailchimpNewsletterSettingsSubmitSuccess = ( siteId, settings ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_NEWSLETTER_SETTINGS_SUBMIT_SUCCESS,
+	siteId,
+	settings
+} );
+
+const mailchimpNewsletterSettingsSubmitFailure = ( siteId, { error } ) => ( {
+	type: WOOCOMMERCE_MAILCHIMP_NEWSLETTER_SETTINGS_SUBMIT_FAILURE,
+	siteId,
+	error
+} );
+
+/**
+ * Triggers a network request to fetch current MailChimp plugin settngs.
+ *
+ * @param  {Number|String} siteId        Jetpack site ID
+ * @return {Function}                    Action thunk
+ */
+export const requestSettings = ( siteId ) => ( dispatch, getState ) => {
+	const state = getState();
+	if ( ! siteId ) {
+		siteId = getSelectedSiteId( state );
+	}
+
+	dispatch( mailchimpSettingsRequest( siteId ) );
+
+	return request( siteId ).get( 'mailchimp' )
+		.then( settings => {
+			dispatch( mailchimpSettingsRequestSuccess( siteId, settings ) );
+		} )
+		.catch( error => {
+			dispatch( mailchimpSettingsRequestFailure( siteId, error ) );
+		} );
+};
+
+/**
+ * Triggers a network request to set MailChimp api key in MailChimp plugin settings.
+ *
+ * @param  {Number|String} siteId        Jetpack site ID
+ * @param  {String}        apiKey        MailChimp api key
+ * @return {Function}                    Action thunk
+ */
+export const submitMailChimpApiKey = ( siteId, apiKey ) => ( dispatch, getState ) => {
+	const state = getState();
+	if ( ! siteId ) {
```

Comment:
Should we return here if `apiKey` is null or not defined? Also I see the simplicity of not having to send a `siteId` but also could this lead to potentially setting the wrong apiKey on a site if the selected site changes somehow?

Reply:
I have seen this `siteId` pattern before in woocommerce Calypso code and even asked about it in the slack, I don't remember the answer but I decided to stick with it. I don't see a problem removing it. 

I agree on the early exit.

Reply:
I think we are doing it in the orders code and possibly a couple other places, and then everywhere else we are not. Definitely worth having a conversation about as a team so we can be consistent.